Formula/kubecfg.rb

```diff
@@ -0,0 +1,48 @@
+class Kubecfg < Formula
+  desc "Manage complex enterprise Kubernetes environments as code"
+  homepage "https://github.com/ksonnet/kubecfg"
+  url "https://github.com/ksonnet/kubecfg/archive/v0.7.2.tar.gz"
+  sha256 "a01de014904af9177bb0ad15a249346d4b7bd6412268d0f83756541aa19fa38c"
+  head "https://github.com/ksonnet/kubecfg.git"
+
+  depends_on "go" => :build
+  depends_on "go-bindata" => :build if build.head?
+
+  def install
+    # Standard gopath shenanigans to keep "go build" happy:
+    sources = buildpath.children - [buildpath/".brew_home"]
+    ENV["GOPATH"] = buildpath
+    srcdir = buildpath/"src/github.com/ksonnet/kubecfg"
+    srcdir.mkpath
+    mv sources, srcdir
+    # The real build steps:
+    cd srcdir do
+      args = []
+      args.push("VERSION=#{version}") unless build.head?
+      system "make", "generate", *args if build.head?
+      system "make", *args
+      system "make", "test", *args if build.head?
```

Comment:
I would move the `make test` task if possible to the `test` hook

Reply:
That's a good idea, but I'm not sure it's possible. The [Formula API](http://www.rubydoc.info/github/Homebrew/brew/master/Formula) says that `buildpath` is only non-nil inside the `install` method. I don't see another way to find the extracted source code, or even a way to provoke its recreation. The man page describes `brew test` as something that
> should generally indicate to the user if something is wrong with the installed formula.